harvester/harvest.py

```diff
@@ -792,30 +769,57 @@ def fill_placeholders(self) -> None:
                 "name": self.harvest_source.get_source_orm().org.name
             }
 
+    def _report_error(self, e):
+        """Report an exception to the database.
+
+        This does not re-raise the exception, it logs it and
+        execution proceeds. If callers want to raise, they need
+        to `raise e` after calling this method.
+        """
+        self.status = "error"
+        log_non_critical_error(
+            repr(e),
+            self.harvest_source.job_id,
+            self.id,
+            e.__class__.__name__,
+            emit_log=False,
+        )
+
     def validate(self) -> None:
+        """Validate a single record.
+
+        Run our jsconschema validator against the raw JSON record or the
+        transformed JSON data.
+
+        Returns True if the record is valid, False if it is not.
+        """
         # TODO: create a different status for transformation exceptions
         # so they aren't confused with validation issues
         logger.info(f"validating {self.identifier}")
-        try:
-            record = (
-                json.loads(self.source_raw)
-                if self.transformed_data is None
-                else self.transformed_data
-            )
+        if self.transformed_data is None:
+            try:
+                record = json.loads(self.source_raw)
+            except json.JSONDecodeError as e:
+                self._report_error(e)
+                self.harvest_source.reporter.update("errored")
+                return False
+        else:
+            record = self.transformed_data
+
+        # save ourselves a second call to is_valid by keeping a flag of
+        # whether we saw any errors
+        valid = True
+        for error in self.harvest_source.validator.iter_errors(record):
+            valid = False
+            self._report_error(error)
 
-            self.harvest_source.validator.validate(record)
-            self.valid = True
+        if valid:
             self.harvest_source.reporter.update("validated")
-        except Exception as e:
-            self.status = "error"
-            self.validation_msg = str(e.message)
-            self.valid = False
+            return True
+        else:
+            # update the reporter only once even with multiple errors
             self.harvest_source.reporter.update("errored")
-            raise ValidationException(
-                repr(e),
-                self.harvest_source.job_id,
-                self.id,
-            )
+            return False
```

Comment:
If we're setting the return value to be truthy instead of throwing an exception, shouldn't we have to change this line here? https://github.com/GSA/datagov-harvester/blob/main/harvester/harvest.py#L657
So the output would need to be parsed and the sync job would be dependent on the output of the validate function.